### DISCUSSION TITLE
Edited node doesn't update

### DISCUSSION THREAD

**Discussion by graddaniel** [xyflow/xyflow#4424] [Q&A]
Hi,
I am trying to edit a node (by finding it in the `nodes` array), by modifying it directly (mainly the `data` object), to rerender the node (display new name, etc).
Upon finishing I am calling the `setNodes` function in various ways:
-By passing the original `nodes` array
-By passing new array, without the modified node, and the modified node appended to it.
-By passing new array, without the modified node, and a new object of the same structure as the modified node, appended to the array.
None of that rerenders the diagram.
NodeResetChange with the original nodes array has no effect.
`updateNodeInternals` has no effect either.

I managed to force some sort of update by calling `onNodesChange` of type "position" with overwritten coordinates, but that obviously repositions the nodes.
When I deselect the node (I select it before editing it) it updates as expected, but I'd like to keep the selection.

Some sort of NodeUpdateChange would work best.

Any  idea how can I achieve that with the current API?

**Comment by bcakmakoglu** [comment]
Can you provide a reproduction of the issue or at least some code?
Otherwise it's hard to tell what you're doing and why it's not working.

**Comment by graddaniel** [comment]
I've just updated to v12, and the issue is gone.
`setNodes([...otherNodes, {...nodeToUpdate}]);`
does the job
